src/libcore/any.rs

```diff
@@ -450,3 +450,22 @@ impl TypeId {
         }
     }
 }
+
+/// Returns the name of a type as a string.
+///
+/// # Note
+///
+/// This is intended for diagnostic use. The exact contents and format of the
+/// string are not specified, other than being a best-effort description of
+/// the type. For example, `type_name::<Option<String>>()` could return the
+/// string `"Option<String>"` or
+/// `"core::option::Option<alloc::string::String>"`, but not `"foobar"`.
```

Comment:
I think this should also mention that the exact output can possibly change between compiler versions without this being considered a breaking change.

Reply:
Done!